blocks/library/table2/index.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { registerBlock, query as hpq } from '../../api';
+import TableBlock from './table-block';
+
+const { children } = hpq;
+
+/**
+ * Returns an attribute setter with behavior that if the target value is
+ * already the assigned attribute value, it will be set to undefined.
+ *
+ * @param  {string}   align Alignment value
+ * @return {Function}       Attribute setter
+ */
+function toggleAlignment( align ) {
+	return ( attributes, setAttributes ) => {
+		const nextAlign = attributes.align === align ? undefined : align;
+		setAttributes( { align: nextAlign } );
+	};
+}
+
+registerBlock( 'core/table2', {
+	title: wp.i18n.__( 'Table2' ),
+	icon: 'editor-table',
+	category: 'common',
```

Comment:
It seems a bit strange from a semantic point of view (as we presumably want to discourage people using tables for just formatting as opposed to actual tabular data) however I have made it consistent with the other table.